File: 04.Simulation模拟/leetcode_3751_totalWaviness.py
class Solution:
    def totalWaviness(self, num1: int, num2: int) -> int:
        def calc(v):
            s = str(v)
            ans = 0
            for i in range(1, len(s)-1):
                if s[i-1]<s[i] and s[i]>s[i+1]:
                    ans += 1
                if s[i-1]>s[i] and s[i]<s[i+1]:
                    ans += 1
            return ans
        res = 0
        for i in range(num1, num2+1):
            res += calc(i)
        return res

if __name__ == "__main__":
    s = Solution()
    num1, num2 = 120, 130
    print(s.totalWaviness(num1, num2))

File: 04.Simulation模拟/leetcode_3751_totalWaviness.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    int totalWaviness(int num1, int num2) {
        int res = 0;
        for (int i=num1; i<=num2; i++){
            string s = to_string(i);
            for (int j=1; j<s.size()-1; j++){
                if ((s[j-1] < s[j] && s[j+1] < s[j]) || (s[j-1] > s[j] && s[j+1] > s[j])){
                    res++;
                }
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    int num1 = 120, num2 = 130;
    cout << s.totalWaviness(num1, num2) << endl;
}
